### PR TITLE
Make `ssc run` respect headless config option

### PR DESCRIPTION
**Pull request by chicoxyzzy**

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
Open question: how to handle headless runners?

File: src/cli/cli.cc [hunk 1]
```diff
@@ -2564,6 +2564,9 @@ int main (const int argc, const char* argv[]) {
     if (isForDesktop) {
       fs::create_directories(paths.platformSpecificOutputPath / "include");
       writeFile(paths.platformSpecificOutputPath / "include" / "user-config-bytes.hh", settings["ini_code"]);
+
+      auto backed = String("const bool __headless = ") + (flagHeadless ? "true" : "false") + ";\n";
```
Comment (jwerle) [inline src/cli/cli.cc:2568]: Should this be `baked`? 🧁

File: src/desktop/main.cc [hunk 2]
```diff
@@ -83,7 +83,7 @@ MAIN {
 
   bool isCommandMode = false;
   bool isReadingStdin = false;
-  bool isHeadless = false;
+  bool isHeadless = isBackedHeadless() || false;
```
Comment (jwerle) [inline src/desktop/main.cc:86]: Same here

File: src/init.cc [hunk 3]
```diff
@@ -14,6 +14,11 @@ namespace SSC {
     ));
   }
 
+  const bool isBackedHeadless () {
+    #include "backed-vars.hh" // NOLINT
```
Comment (jwerle) [inline src/init.cc:18]: 🙏🏼

**Review by jwerle — APPROVED** [review]

**Pull request merged**